nucypher/cli/commands/alice.py

```diff
@@ -387,6 +388,14 @@ def grant(general_config,
 
     ALICE = character_options.create_character(emitter, config_file, general_config.json_ipc)
 
+    # Input validation
+    if ALICE.federated_only:
+        if any((value, rate)):
+            raise click.BadOptionUsage(option_name="--value, --rate",
+                                       message="Can't use --value or --rate with a federated Alice.")
+    elif not (bool(value) ^ bool(rate)):
+        raise click.BadOptionUsage(option_name="--rate", message="Can't use --value if using --rate")
```

Comment:
Since we are saying "Can't use --value...", should this line instead be (?):
```suggestion
        raise click.BadOptionUsage(option_name="--value", message="Can't use --value if using --rate")
```

or alternatively

```suggestion
        raise click.BadOptionUsage(option_name="--rate", message="Can't use --rate if using --value")
```